sdk/python/kfp/dsl/_metadata.py

```diff
@@ -70,7 +70,10 @@ def serialize(self):
 
   @staticmethod
   def deserialize(payload):
-    # If the payload is a string of a dict serialization, convert it back to a dict
+    '''deserialize expects two types of input: dict and str
+    1) If the payload is a string of a dict serialization, convert it back to a dict
+    2) If the payload is a string, the type is named as such with no properties.
+    3) If the payload is a dict, the type name and properties are extracted. '''
     try:
       import ast
       payload = ast.literal_eval(payload)
```

Comment:
I just saw this and I thing this is not a good thing....

Reply:
Please suggest. we can send another PR since this PR is simply to add more comments here.